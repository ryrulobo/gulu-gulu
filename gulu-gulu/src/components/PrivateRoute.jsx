import { Navigate, Outlet } from "react-router-dom";

export default function PrivateRoute() {
  const access_token = localStorage.getItem("access_token");
  if (!access_token) {
    return <Navigate to={"/"} />;
  }
  return (
    <div data-testid="privateRoute">
      <Outlet />
    </div>
  );
}
