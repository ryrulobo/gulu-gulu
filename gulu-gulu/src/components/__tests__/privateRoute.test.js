import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import PrivateRoute from "../PrivateRoute";

afterEach(() => {
  cleanup();
});

localStorage.setItem("access_token", "dummy_token");
const access_token = localStorage.getItem("access_token");

test("Render private route", () => {
  render(
    <BrowserRouter>
      <PrivateRoute access_token={access_token} />
    </BrowserRouter>
  );
  const privateRouteElement = screen.getByTestId("privateRoute");
  expect(privateRouteElement).toBeInTheDocument();
});

test("Matches snapshots", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <PrivateRoute access_token={access_token} />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
