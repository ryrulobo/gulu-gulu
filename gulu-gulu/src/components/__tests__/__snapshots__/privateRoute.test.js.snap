// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1`] = `
<div
  data-testid="privateRoute"
/>
`;
